### PR TITLE
[Storage] Support preserverPerserve File Attributes

### PR DESCRIPTION
**Pull request by blueww**
## Description

This is to support preserver File SMB attributes when upload/download Azure File.
The design has pass review as part of https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/370

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage/File/Cmdlet/GetAzureStorageFileContent.cs [hunk 1]
```diff
@@ -201,7 +202,8 @@ public override void ExecuteCmdlet()
                             targetFile,
                             new DownloadOptions
                             {
-                                DisableContentMD5Validation = !this.CheckMd5
+                                DisableContentMD5Validation = !this.CheckMd5,
+                                PreserveSMBAttributes = context.PreserveSMBAttribute.IsPresent ? true : false
```
[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/File/Cmdlet/GetAzureStorageFileContent.cs]: Fixed.

File: src/Storage/Storage/File/Cmdlet/GetAzureStorageFileContent.cs [hunk 2]
```diff
@@ -221,5 +223,15 @@ public override void ExecuteCmdlet()
                 DoEndProcessing();
             }
         }
+        public object GetDynamicParameters()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```
[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/File/Cmdlet/GetAzureStorageFileContent.cs]: Fixed.

File: src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs [hunk 3]
```diff
@@ -227,5 +232,16 @@ private async Task<CloudFile> BuildCloudFileInstanceFromPathAsync(string default
                 return baseDirectory.GetFileReferenceByPath(path);
             }
         }
+
+        public object GetDynamicParameters()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```
[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs]: Fixed.

File: src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs [hunk 4]
```diff
@@ -129,7 +131,10 @@ public override void ExecuteCmdlet()
                     this.TransferManager.UploadAsync(
                             localFile.FullName,
                             cloudFileToBeUploaded,
-                            null,
+                            new UploadOptions
+                            {
+                                PreserveSMBAttributes = context.PreserveSMBAttribute.IsPresent ? true : false
```
[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs]: Good point. Fixed.

File: src/Storage/Storage/File/Cmdlet/GetAzureStorageFileContent.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/File/Cmdlet/GetAzureStorageFileContent.cs]: Good point. Fixed.

**Comment by markcowl** [comment]
All checks pass: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=110615&view=results
https://dev.azure.com/azure-sdk/public/_build/results?buildId=110613&view=results
https://dev.azure.com/azure-sdk/public/_build/results?buildId=110614&view=results

**Comment by blueww** [comment]
@markcowl 
Thanks for helping review and merge it!

**Pull request merged**